api/lib/mno_enterprise/concerns/controllers/jpi/v1/app_instances_controller.rb

```diff
@@ -13,31 +13,33 @@ module MnoEnterprise::Concerns::Controllers::Jpi::V1::AppInstancesController
   #==================================================================
   # Instance methods
   #==================================================================
-  # GET /mnoe/jpi/v1/organization/1/apps.json?timestamp=151452452345
+  # GET /mnoe/jpi/v1/organization/1/apps.json?tg=151452452345
   def index
-    @app_instances = parent_organization.app_instances.active.where("updated_at.gt" => Time.at(timestamp)).select do |i|
-      # force owner assignment to avoid a refetch in ability can?(:access,i)
-      i.owner = parent_organization
+    @app_instances = Mno::AppInstance.includes(:app, :owner).where(owner_id: parent_organization.id, status: Mno::AppInstance::ACTIVE_STATUSES).to_a.select do |i|
       can?(:access,i)
     end
   end
 
   # POST /mnoe/jpi/v1/organization/1/app_instances
   def create
     authorize! :manage_app_instances, parent_organization
-    app_instance = parent_organization.app_instances.create(product: params[:nid])
-    MnoEnterprise::EventLogger.info('app_add', current_user.id, 'App added', app_instance)
-    head :created
+    app_instance = parent_organization.provision_app_instance(params[:nid])
+    if app_instance.errors.any?
+      render json: app_instance.errors, status: :bad_request
+    else
+      MnoEnterprise::EventLogger.info('app_add', current_user.id, 'App added', app_instance.first)
```

Comment:
This is the way of working of json_api_client, everything returned is always an array.